.github/workflows/unit_test.yml

```diff
@@ -0,0 +1,48 @@
+# This will automatically use the sbt version in project/build.properties
+name: Unit tests
+
+on:
+  push:
+    branches: [ develop ]
+  pull_request:
+    branches: [ develop ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up mysql docker container
+      run: |
+         docker rm -f mysql || true
+         docker pull mysql/mysql-server:5.6
+         docker run -p 3307:3306 --name mysql -e MYSQL_ROOT_PASSWORD=leonardo-test -e MYSQL_USER=leonardo-test -e MYSQL_PASSWORD=leonardo-test -e MYSQL_DATABASE=leotestdb -d mysql/mysql-server:5.6
+
+    - name: Set up JDK 1.8
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.8
+    - name: Compile and check scalafmt
+      env:
+        JAVA_OPTS: -Xmx3G
+        JVM_OPTS:  -Xmx3G
+      run: sbt -Denv.type=test clean "test:compile" scalafmtCheckAll
+
+    - name: Run tests
+      env:
+        JAVA_OPTS: -Xmx3G
+        JVM_OPTS:  -Xmx3G
+      run: |
+          sbt -Denv.type=test -Dmysql.host=localhost -Dmysql.port=3307 "project http" coverage test coverageReport
+          sbt -Denv.type=test "project core" coverage test coverageReport
```

Comment:
I would run tests for `core` before `http`

Reply:
one thing I'm not sure how things work is how these codecov reports get published in travis...I'm thinking we'll need to do something for that in github workflow? (at least I think codecov needs some token defined?)